services/scrape-test/scrape_media_url.py

```diff
@@ -0,0 +1,51 @@
+from playwright.sync_api import BrowserContext
+from typing import Optional
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def scrape_youtube_link(url: str, context: BrowserContext) -> Optional[str]:
+    """
+    Scrapes the given URL for an embedded YouTube link by inspecting the frames.
+
+    Args:
+        url (str): The URL to scrape.
+        context (BrowserContext): The Playwright browser context to use for scraping.
+
+    Returns:
+        Optional[str]: The YouTube link if found, "timeout_error" on timeout, or None if no link is found.
+    """
+    logger.info(f"Starting to scrape for YouTube link at URL: {url}")
+    page = context.new_page()
+    try:
+        page.goto(url, timeout=60000)
```

Comment:
I wouldn't use that big timeouts. if for some reason, that page is broken it just blocks your process. It's better to have shorter timeout and store unsuccessful url's somewhere, and try to go back to them again

Reply:
I changed it to 20 seconds for now, after testing more we will take it down another level